src/cmd/services/m3dbnode/config/bootstrap.go

```diff
@@ -70,18 +74,46 @@ type BootstrapConfiguration struct {
 type BootstrapFilesystemConfiguration struct {
 	// NumProcessorsPerCPU is the number of processors per CPU.
 	NumProcessorsPerCPU float64 `yaml:"numProcessorsPerCPU" validate:"min=0.0"`
+
+	// Migrations configuration
+	Migrations *BootstrapMigrations `yaml:"migrations"`
 }
 
 func (c BootstrapFilesystemConfiguration) numCPUs() int {
 	return int(math.Ceil(float64(c.NumProcessorsPerCPU * float64(runtime.NumCPU()))))
 }
 
+func (c BootstrapFilesystemConfiguration) migrations() BootstrapMigrations {
+	if cfg := c.Migrations; cfg != nil {
+		return *cfg
+	}
+	return BootstrapMigrations{Concurrency: defaultMigrationConcurrency}
+}
+
 func newDefaultBootstrapFilesystemConfiguration() BootstrapFilesystemConfiguration {
 	return BootstrapFilesystemConfiguration{
 		NumProcessorsPerCPU: defaultNumProcessorsPerCPU,
+		Migrations:          &BootstrapMigrations{},
 	}
 }
 
+// BootstrapMigrations specifies configuration for data migrations during bootstrapping
+type BootstrapMigrations struct {
+	// ToVersion1_1Task indicates that we should attempt to upgrade filesets to
+	// what’s expected of 1.1 files
+	ToVersion1_1 bool `yaml:"toVersion1_1"`
```

Comment:
This may be better as an enum to make it more extensible, , similar to how we do MetricsType here: https://github.com/m3db/m3/blob/master/src/query/storage/m3/storagemetadata/config.go

Reply:
Agreed, maybe it can be an enum such as:
```type
type MigrateVersion uint

const (
  MigrateVersionNone MigrateVersion = iota
  MigrateVersion_1_1
  // .. we add values here later
)
```

Can make the string version of "MigrateVersion_1_1" be "1.1" too so it's clear which version you are upgrading to:
```go
var (
	validMigrateVersions = []MigrateVersion{
		MigrateVersionNone,
		MigrateVersion_1_1,
	}
)

func (v MigrateVersion) String() string {
	switch v {
	case MigrateVersionNone:
		return "none"
	case MigrateVersion_1_1:
		return "1.1"
	default:
		return "unknown"
	}
}

func ParseMigrateVersion(str string) (MigrateVersion, error) {
	for _, valid := range validMigrateVersions {
		if str == valid.String() {
			return valid, nil
		}
	}

	return 0, fmt.Errorf("unrecognized migrate version: %v", str)
}
```

Reply:
I like this. Will convert this to an enum throughout